zap/src/main/java/org/zaproxy/zap/extension/api/ContextAPI.java

```diff
@@ -465,6 +466,18 @@ private ApiResponse buildResponseFromContext(Context c) {
         fields.put("inScope", Boolean.toString(c.isInScope()));
         fields.put("excludeRegexs", jsonEncodeList(c.getExcludeFromContextRegexs()));
         fields.put("includeRegexs", jsonEncodeList(c.getIncludeInContextRegexs()));
+        fields.put(
+                "includedTechnologyList",
+                jsonEncodeList(
```

Comment:
It would be better if this returned an actual list, instead of a string.

Reply:
instead of returning only tech name(`toString`), returning c.getTechSet().getExcludeTech() entire list? Am I right?

Reply:
I have modified the code to return `Tech` object instead of just `Tech::toString()`/`Tech::uiName()`

Reply:
It's not about the contents but the type that's returned, using the `toString` was fine. If you check the returned JSON object the `includedTechnologies` is a string not a list, that means the API callers need to further parse it to get the actual techs.

Instead of converting the list to a string keep it as list and return a custom `ApiResponseSet`, e.g.:
```java
private static class ContextApiResponseSet<T> extends ApiResponseSet<T> {

    ContextApiResponseSet(String name, Map<String, T> values) {
        super(name, values);
    }

    @Override
    public void toXML(Document doc, Element parent) {
        parent.setAttribute("type", "set");
        for (Map.Entry<String, T> val : getValues().entrySet()) {
            Element el = doc.createElement(val.getKey());
            if ("includedTechnologies".equals(val.getKey())
                    || "excludedTechnologies".equals(val.getKey())) {
                el.setAttribute("type", "list");
                @SuppressWarnings("unchecked")
                List<String> techs = (List<String>) val.getValue();
                for (String tech : techs) {
                    Element elTech = doc.createElement("tech");
                    elTech.appendChild(
                            doc.createTextNode(XMLStringUtil.escapeControlChrs(tech)));

                    el.appendChild(elTech);
                }
            } else {
                String textValue = val.getValue() == null ? "" : val.getValue().toString();
                Text text = doc.createTextNode(XMLStringUtil.escapeControlChrs(textValue));
                el.appendChild(text);
            }
            parent.appendChild(el);
        }
    }
}
```

Reply:
Got your point. Thanks for explaining. I'll correct the code and push again.

Reply:
I have changed the code as per your suggestions. Actually, `APIResponseSet.toJSON()`, only converts to Json friendly string (with backslashes) only if we provide string. In current Context View API had a map of `Map<String,String>`. I changed it to `Map<String, Object>`.  After  the change, now the context API returns `Tech` as List. One doubt, I checked the response of Context View API.  Th keys such as "includeRegexes", "postParameterParserConfig" and others, still in string. I didn't touch them. Do We have to change them as well?
`{
  "context": {
    "excludeRegexs": "[]",
    "loggedInPattern": "",
    "urlParameterParserClass": "org.zaproxy.zap.model.StandardParameterParser",
    "description": "",
    "includeRegexs": "[]",
    "checkingStrategy": "EACH_RESP",
    "inScope": "true",
    "authenticationDetectionMethodId": "0",
    "loggedOutPattern": "",
    "postParameterParserConfig": "{\"kvps\":\"&\",\"kvs\":\"=\",\"struct\":[]}",
    "excludedTechnologies": [],
    "name": "Default Context",
    "includedTechnologies": [
      "Db",
      "Db.CouchDB",
      "Db.Firebird",
      "Db.HypersonicSQL",
      "Db.IBM DB2",
      "Db.Microsoft Access",
      "Db.Microsoft SQL Server",
      "Db.MongoDB",
      "Db.MySQL",
      "Db.Oracle",
      "Db.PostgreSQL",
      "Db.SAP MaxDB",
      "Db.SQLite",
      "Db.Sybase",
      "Language",
      "Language.ASP",
      "Language.C",
      "Language.JSP/Servlet",
      "Language.Java",
      "Language.Java.Spring",
      "Language.JavaScript",
      "Language.PHP",
      "Language.Python",
      "Language.Ruby",
      "Language.XML",
      "OS",
      "OS.Linux",
      "OS.MacOS",
      "OS.Windows",
      "SCM",
      "SCM.Git",
      "SCM.SVN",
      "WS",
      "WS.Apache",
      "WS.IIS",
      "WS.Tomcat"
    ],
    "urlParameterParserConfig": "{\"kvps\":\"&\",\"kvs\":\"=\",\"struct\":[]}",
    "id": "1",
    "authType": "Manual Authentication",
    "postParameterParserClass": "org.zaproxy.zap.model.StandardParameterParser"
  }
}`

Reply:
I pushed the changes to my branch.

Reply:
No, other fields should be kept as is otherwise it would break existing callers that are already consuming those fields.

It needs to use the custom `ApiResponseSet` otherwise the XML response returns the fields `includedTechnologies` and `excludedTechnologies` incorrectly.

Reply:
I have changed the code with `ContextApiResponseSet`